codec/src/main/java/io/netty/handler/codec/compression/JdkZlibDecoder.java

```diff
@@ -30,6 +38,36 @@
  * Decompress a {@link ByteBuf} using the inflate algorithm.
  */
 public class JdkZlibDecoder extends ZlibDecoder {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(JdkZlibDecoder.class);
+    private static final Method INFLATER_SET_INPUT_BYTEBUFFER;
+
+    static {
+        Method inflaterSetInput = null;
+        if (PlatformDependent.javaVersion() >= 11) {
+            final Object maybeException =
+                    AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                        @Override
+                        public Object run() {
+                            try {
+                                return Inflater.class.getDeclaredMethod("setInput", ByteBuffer.class);
+                            } catch (NoSuchMethodException e) {
+                                return e;
+                            } catch (SecurityException e) {
+                                return e;
+                            }
+                        }
+                    });
+            if (maybeException instanceof Method) {
+                inflaterSetInput = (Method) maybeException;
+                logger.debug("Inflater.setInput(ByteBuffer): available");
+            } else {
+                logger.debug("Inflater.setInput(ByteBuffer): available");
```

Comment:
@fabienrenaud this is not correct... also include the exception here as well

Reply:
fixed + done.